pkg/bmt/inclusionproof_test.go

```diff
@@ -0,0 +1,106 @@
+package bmt_test
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/bmt"
+	"github.com/ethersphere/bee/pkg/bmt/reference"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func TestInclusionProofs(t *testing.T) {
+
+	testDatas := []struct {
```

Comment:
i believe that due to the importance of this feature, it would be prudent to have a manually calculated test that checks every byte of the proof is the way it should be. i.e. generating something and then silently verifying it goes only half-way. i'd expect to have tests that cover:
- what happens when the proof is on the left end of the tree
- what happens when its on the right
- middle
- encrypted chunks work correctly
- ideally you'll also calculate "by-hand" the tree with `keccak()` to see that the results add up

Reply:
So I added the cases now with hard-coded values which were calculated manually.

I am not sure if we need to add separate case for encrypted data though. The hasher doesnt seem to care if the payload is encrypted. We already have tests with random data below. So I think we should be covered now.